tritium-metrics/src/main/java/com/palantir/tritium/event/metrics/TaggedMetricsServiceInvocationEventHandler.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.event.metrics;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.palantir.logsafe.UnsafeArg;
+import com.palantir.tritium.event.AbstractInvocationEventHandler;
+import com.palantir.tritium.event.DefaultInvocationContext;
+import com.palantir.tritium.event.InstrumentationProperties;
+import com.palantir.tritium.event.InvocationContext;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.lang.reflect.Method;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An implementation of {@link AbstractInvocationEventHandler} whose purpose is to provide tagged metrics for
+ * classes which look like services.
+ *
+ * Specifically, this class will generate metrics with the following parameters:
+ *
+ * <ul>
+ *     <li>Metric Name: the service name supplied to the constructor</li>
+ *     <li>Tag - service-name: The simple name of the invoked class</li>
+ *     <li>Tag - endpoint: The name of the method that was invoked</li>
+ *     <li>Tag - cause: When an error is hit, this will be filled with the full class name of the cause.</li>
+ * </ul>
+ */
+public class TaggedMetricsServiceInvocationEventHandler extends AbstractInvocationEventHandler<InvocationContext> {
+
+    private static final Logger logger = LoggerFactory.getLogger(TaggedMetricsServiceInvocationEventHandler.class);
+
+    private final TaggedMetricRegistry taggedMetricRegistry;
+    private final String serviceName;
+
+    public TaggedMetricsServiceInvocationEventHandler(
+            TaggedMetricRegistry taggedMetricRegistry,
+            String serviceName) {
+        super(getEnabledSupplier(serviceName));
+        this.taggedMetricRegistry = checkNotNull(taggedMetricRegistry, "metricRegistry");
+        this.serviceName = checkNotNull(serviceName, "serviceName");
+    }
+
+    private static String failuresMetricName() {
+        return "failures";
```

Comment:
any reason not to define a constant instead?

Reply:
Done